### ISSUE TITLE
`provider_refresh_token` is missing for linkedin auth

### ISSUE THREAD

**Issue by devenbhooshan** [supabase/supabase#15161]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

`provider_refresh_token` is null in session object obtained from supabase.auth.getSession() for LinkedIn Auth.
## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create an app that supports login with LinkedIn 
2. Log-in in that app
3. `provider_refresh_token` is missing in the session object

## Expected behavior

The `provider_refresh_token` should be available and should be non empty.

## Screenshots
![Group 7](https://github.com/supabase/supabase/assets/2411744/57e0ce25-920f-49fe-8554-0046006fb1eb)

## System information

- OS: macOS
- Browser: chrome
- Version of supabase-js: 2.20.0
- Version of Node.js: v16.19.1

## Additional context

None

**Comment by aleciavogel** [comment]
Can you provide an example of how you're signing in? This might be a duplicate of #13266

**Comment by devenbhooshan** [comment]
Hi @aleciavogel 
I am posting the code snippets here. Do check it out. 


```
async function signInWithLinkedIn() {
      // get redirect url from query params
      const urlParams = new URLSearchParams(window.location.search);
      const redirectUrl = urlParams.get('redirectUrl') || '/';

      login(redirectUrl).catch((error) => {
        console.error(error);
        // Todo: handle errors
      });
    }
    
export async function login(redirectUrl){
  await supabase.auth.signInWithOAuth({
    provider: 'linkedin',
    options: {
      scopes: 'r_liteprofile r_emailaddress w_member_social',
      redirectTo: window.location.origin + '/auth/linkedin/callback?redirectUrl=' + redirectUrl,
    },
  });
}

----

<button title=""
          onClick={signInWithLinkedIn}
          className="inline-flex items-center justify-center gap-2 px-6 py-3.5 text-base font-semibold text-white transition-all duration-150 bg-blue-500 rounded-full shadow-md hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500"
          >
          <svg aria-hidden="true" className="w-6 h-6" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"
            fill="currentColor">
            <path
              d="M20 3H4a1 1 0 0 0-1 1v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1zM8.339 18.337H5.667v-8.59h2.672v8.59zM7.003 8.574a1.548 1.548 0 1 1 0-3.096 1.548 1.548 0 0 1 0 3.096zm11.335 9.763h-2.669V14.16c0-.996-.018-2.277-1.388-2.277-1.39 0-1.601 1.086-1.601 2.207v4.248h-2.667v-8.59h2.56v1.174h.037c.355-.675 1.227-1.387 2.524-1.387 2.704 0 3.203 1.778 3.203 4.092v4.71z">
            </path>
          </svg>
          Login with LinkedIn
        </button>
``

**Comment by aleciavogel** [comment]
Are you an approved LinkedIn Marketing Developer Platform partner?

[Per their docs:](https://learn.microsoft.com/en-us/linkedin/shared/authentication/programmatic-refresh-tokens?context=linkedin%2Fcontext)

> LinkedIn supports programmatic refresh tokens for all approved Marketing Developer Platform (MDP) partners.

Instead, they suggest the following:

> Refreshing an access token is a seamless user experience. To refresh an access token, go through the [authorization process](https://learn.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow?tabs=HTTPS1#step-2-request-an-authorization-code) again to fetch a new token. This time however, in the refresh workflow, the authorization screen is bypassed, and the member is redirected to your redirect URL, provided the following conditions are met:
>
> * The member is still logged into [www.linkedin.com](https://www.linkedin.com/)
> * The member's current access token has not expired

I saw somewhere that the access token is valid for 60 days by default, but I'm not sure if that's MDP-specific.

**Comment by devenbhooshan** [comment]
> Are you an approved LinkedIn Marketing Developer Platform partner?
No. Sorry, how did I miss this :( 

> I saw somewhere that the access token is valid for 60 days by default, but I'm not sure if that's MDP-specific.

Yes, the access token is valid for 60 days. 

Looks like the issue is not with Supabase. I am closing this ticket. 

Thanks, @aleciavogel

**Comment by aleciavogel** [comment]
Glad I could help. Your issue popped up when I was blocked by something similar, except with Google 🥲

Hope it all works out!
